storageapi/src/vespa/storageapi/messageapi/storagemessage.cpp

```diff
@@ -143,90 +143,68 @@ std::ostream & operator << (std::ostream & os, const StorageMessageAddress & add
     return os << addr.toString();
 }
 
-static vespalib::string
-createAddress(vespalib::stringref cluster, const lib::NodeType& type, uint16_t index)
-{
+namespace {
+
+vespalib::string
+createAddress(vespalib::stringref cluster, const lib::NodeType &type, uint16_t index) {
     vespalib::asciistream os;
     os << STORAGEADDRESS_PREFIX << cluster << '/' << type.toString() << '/' << index << "/default";
     return os.str();
 }
 
-size_t
-calculate_node_hash(const lib::NodeType& type, uint16_t index)
-{
-    uint16_t buf[] = { type, index };
-    return vespalib::hashValue(&buf, sizeof(buf));
+uint32_t
+calculate_node_hash(const lib::NodeType &type, uint16_t index) {
+    uint16_t buf[] = {type, index};
+    size_t hash =  vespalib::hashValue(&buf, sizeof(buf));
+    return uint32_t(hash & 0xffffffffl) ^ uint32_t(hash >> 32);
```

Comment:
@vekterli should look at this.

Reply:
The entropy quality of the lower and higher 32 bits of the XXH3 output should be pretty much the same, but mixing them together doesn't hurt either. If we didn't mix, we probably should have added an explicit `uint32_t` cast anyway to avoid any latent truncation warnings from the compiler.